src/App/Fossa/Analyze.hs

```diff
@@ -332,33 +301,40 @@ analyze ::
   , Has Debug sig m
   , Has Exec sig m
   , Has ReadFS sig m
-  , Has (Reader AnalyzeExperimentalPreferences) sig m
-  , MonadIO m
+  , Has (Reader StandardAnalyzeConfig) sig m
   ) =>
-  BaseDir ->
-  ScanDestination ->
-  OverrideProject ->
-  Flag UnpackArchives ->
-  Flag JsonOutput ->
-  Flag IncludeAll ->
-  ModeOptions ->
-  AllFilters ->
   m ()
-analyze (BaseDir basedir) destination override unpackArchives jsonOutput includeAll ModeOptions{..} filters = Diag.context "fossa-analyze" $ do
+analyze = Diag.context "fossa-analyze" $ do
   capabilities <- sendIO getNumCapabilities
 
+  -- TODO: refactor other code to use config reader, rather than passing
+  -- values directly and defining them upfront like this.
+  cfg <- ask
```

Comment:
This is isomorphic to passing in the config as an argument to `analyze`. Let's do that instead, because the `Reader` isn't used anywhere else

Reply:
Yeah, that was from the original pass, but I think that's not nearly as useful now.  I'll change it.